### PR TITLE
3588: Create Access keys, CL and UI

### PR DESCRIPTION
**Pull request by John-VanHook**
**What does this PR do, and why do we need it?**

**Which issue does this PR fix?**

fixes #<ISSUE>

**Special notes for your reviewers**:

### CODE REVIEW

File: docs/docsource/operation/account_operations/create_an_account.rst [hunk 1]
```diff
@@ -34,31 +34,66 @@ not yet set up any accounts:
    click the **Locations** tab to manage account locations and the buckets they
    contain.
    
-Create an Account from the Command Line
+Create an Account From the Command Line
```
Comment (lucieleonard) [inline docs/docsource/operation/account_operations/create_an_account.rst]: We need to create the same title but for the UI procedure above

File: docs/docsource/operation/account_operations/manage_access_keys.rst [hunk 2]
```diff
@@ -0,0 +1,51 @@
+.. _manage_access_keys:
+
+Manage Access Keys
+==================
+
+Creating Access Keys Using the UI
+---------------------------------
+
+#. Log into the |product| UI using previously provisioned credentials.
```
Comment (lucieleonard) [inline docs/docsource/operation/account_operations/manage_access_keys.rst]: I would put a :ref: to the right topic on "previously provisioned credentials"

File: docs/docsource/operation/account_operations/manage_access_keys.rst [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+   .. image:: ../../graphics/sign_into_artesca.PNG
+
+#. Select the **Accounts** tab at the top of the screen.
```
Comment (lucieleonard) [inline docs/docsource/operation/account_operations/manage_access_keys.rst]: I'd say "Click **Accounts** at the top menu." ?

File: docs/docsource/operation/account_operations/manage_access_keys.rst [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+   .. image:: ../../graphics/account_banner.PNG
+
+#. Select blue button  **+Create Access key**
```
Comment (lucieleonard) [inline docs/docsource/operation/account_operations/manage_access_keys.rst]: I'd say "Click **+ Create Access Keys**.

File: docs/docsource/operation/account_operations/manage_access_keys.rst [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+   .. image:: ../../graphics/accounts-page.PNG
+
+#. Select **Continue** on the **Create Root user Access keys** screen.
```
Comment (lucieleonard) [inline docs/docsource/operation/account_operations/manage_access_keys.rst]: Click

File: docs/docsource/operation/account_operations/manage_access_keys.rst [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+   .. image:: ../../graphics/root_user_access_keys_dialog.PNG
+
+#. Save the Secret Access key ID.
```
Comment (lucieleonard) [inline docs/docsource/operation/account_operations/manage_access_keys.rst]: Key (capital letter) + maybe add "by clicking the copy button, and save it in a secure location." (then get rid of the sentence below)

File: docs/docsource/operation/account_operations/manage_access_keys.rst [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+   .. image:: ../../graphics/root_user_access_keys_dialog.PNG
+
+#. Save the Secret Access key ID. 
+   
+   .. image:: ../../graphics/close_button.PNG
+   
+   Save the **Access key ID** in a secure location.
+
+#. Select the **Close** button to finish.
```
Comment (lucieleonard) [inline docs/docsource/operation/account_operations/manage_access_keys.rst]: Click close

File: docs/docsource/operation/account_operations/manage_access_keys.rst [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+Deleting Access Keys Using the UI
+---------------------------------
+
+#. Log into the |product| UI using previously provisioned credentials.
```
Comment (lucieleonard) [inline docs/docsource/operation/account_operations/manage_access_keys.rst]: same comment as above

File: docs/docsource/operation/account_operations/manage_access_keys.rst [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+
+   .. image:: ../../graphics/sign_into_artesca.PNG
+
+#. Navigate to the **Accounts** tab at the top of the screen.
+
+   .. image:: ../../graphics/account_banner.PNG
+
+#. Select the red trash icon to the right of the access key you wish to delete.
```
Comment (lucieleonard) [inline docs/docsource/operation/account_operations/manage_access_keys.rst]: Click

File: docs/docsource/operation/account_operations/manage_access_keys.rst [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+#. Navigate to the **Accounts** tab at the top of the screen.
+
+   .. image:: ../../graphics/account_banner.PNG
+
+#. Select the red trash icon to the right of the access key you wish to delete. 
+
+   .. note::
+
+	   You will not be asked for confirmation. Selecting the trash icon will immediatly delete the key.
```
Comment (lucieleonard) [inline docs/docsource/operation/account_operations/manage_access_keys.rst]: typo: immediately

**Review by thomasdanan — CHANGES_REQUESTED** [review]
@lucieleonard @John-VanHook I have 2 global comments

1. We should have a consistent organization of the information: Currently for each and every operation, you will get the UI way and the CLI way (within the same page). As such (if we are following this logic), generating Account Access Key in CLI should be under Managing Access Key Section
2. I think it does not make sense to put the login page for all operations, I would vote for removing it where it is useless.

I didn't test the procedure but I assume it is working ?

Thanks

Thomas

File: docs/docsource/operation/account_operations/create_an_account.rst [hunk 11]
```diff
@@ -34,31 +34,66 @@ not yet set up any accounts:
    click the **Locations** tab to manage account locations and the buckets they
    contain.
    
-Create an Account from the Command Line
----------------------------------------
+Create an Account Using the Command Line
+----------------------------------------
 
-To create an account from the command line:
+#. Set the admin access key ID and secret access key ID as environment variables:
```
Comment (cathydossantospinto) [inline docs/docsource/operation/account_operations/create_an_account.rst]:
```suggestion
#. Set the admin access key ID and secret access key ID as environment variables.
```

File: docs/docsource/operation/account_operations/create_an_account.rst [hunk 12]
```diff
@@ -34,31 +34,66 @@ not yet set up any accounts:
...
-#. Issue the following variable declarations and commands:
+      ADMIN_ACCESS_KEY_ID=$(kubectl --kubeconfig=/etc/kubernetes/admin.conf 
+      get secret artesca-data-management-vault-admin-creds.v1 -n zenko -o jsonpath='{.data.accessKey}' | base64 -d)
+
+   .. code::
+
+      ADMIN_SECRET_ACCESS_KEY=$(kubectl --kubeconfig=/etc/kubernetes/admin.conf 
+      get secret artesca-data-management-vault-admin-creds.v1 -n zenko -o jsonpath='{.data.secretKey}' | base64 -d)
+
+#. Create the Account
```
Comment (cathydossantospinto) [inline docs/docsource/operation/account_operations/create_an_account.rst]:
```suggestion
#. Create the account.
```

File: docs/docsource/operation/account_operations/create_an_account.rst [hunk 13]
```diff
@@ -34,31 +34,66 @@ not yet set up any accounts:
...
+   .. code::
+
+      ADMIN_SECRET_ACCESS_KEY=$(kubectl --kubeconfig=/etc/kubernetes/admin.conf 
+      get secret artesca-data-management-vault-admin-creds.v1 -n zenko -o jsonpath='{.data.secretKey}' | base64 -d)
+
+#. Create the Account 
+
+   .. note:: 
+
+      In the following example, account name is ``account1`` and email is ``account1@scality.local``
```
Comment (cathydossantospinto) [inline docs/docsource/operation/account_operations/create_an_account.rst]:
```suggestion
      In the following example, the account name is ``account1`` and the email is ``account1@scality.local``.
```

File: docs/docsource/operation/account_operations/manage_access_keys.rst [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+
+#. Log into the |product| UI using previously provisioned credentials (refer to :ref:`Create an Account`).
+
+   .. image:: ../../graphics/sign_into_artesca.PNG
+
+#. Click **Accounts** on the top menu.
+
+   .. image:: ../../graphics/account_banner.PNG
+
+#. Click **+Create Access key**
```
Comment (cathydossantospinto) [inline docs/docsource/operation/account_operations/manage_access_keys.rst]:
```suggestion
#. Click **+ Create Access key**.
```

File: docs/docsource/operation/account_operations/manage_access_keys.rst [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+
+   .. image:: ../../graphics/accounts-page.PNG
+
+#. Click **Continue** on the **Create Root user Access keys** screen.
```
Comment (cathydossantospinto) [inline docs/docsource/operation/account_operations/manage_access_keys.rst]:
```suggestion
#. Click **-> Continue** on the **Create Root user Access keys** screen.
```

File: docs/docsource/operation/account_operations/manage_access_keys.rst [hunk 16]
```diff
@@ -0,0 +1,49 @@
...
+
+   .. image:: ../../graphics/root_user_access_keys_dialog.PNG
+
+#. Save the Secret Access Key ID by clicking the copy button, and save it in a secure location
```
Comment (cathydossantospinto) [inline docs/docsource/operation/account_operations/manage_access_keys.rst]:
```suggestion
#. Save the Secret Access key ID by clicking the copy button, and save it in a secure location.
```

File: docs/docsource/operation/account_operations/manage_access_keys.rst [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+
+#. Click **Continue** on the **Create Root user Access keys** screen.
+
+   .. image:: ../../graphics/root_user_access_keys_dialog.PNG
+
+#. Save the Secret Access Key ID by clicking the copy button, and save it in a secure location 
+   
+   .. image:: ../../graphics/close_button.PNG
+
+#. Close **Close**.
```
Comment (cathydossantospinto) [inline docs/docsource/operation/account_operations/manage_access_keys.rst]:
```suggestion
#. Click **Close** once you saved the Secret Access key ID.
```

File: docs/docsource/operation/account_operations/manage_access_keys.rst [hunk 18]
```diff
@@ -0,0 +1,49 @@
...
+
+   .. image:: ../../graphics/sign_into_artesca.PNG
+
+#. Click **Accounts** on the top menu.
+
+   .. image:: ../../graphics/account_banner.PNG
+
+#. Click the red trash icon to the right of the access key you wish to delete. 
+
+   .. note::
```
Comment (cathydossantospinto) [inline docs/docsource/operation/account_operations/manage_access_keys.rst]: Maybe put this in an important. I feel like a note isn't enough.

Comment (John-VanHook) [reply docs/docsource/operation/account_operations/manage_access_keys.rst]: I agree

**Comment by John-VanHook** [comment]
> @lucieleonard @John-VanHook I have 2 global comments
> 
> 1. We should have a consistent organization of the information: Currently for each and every operation, you will get the UI way and the CLI way (within the same page). As such (if we are following this logic), generating Account Access Key in CLI should be under Managing Access Key Section
> 2. I think it does not make sense to put the login page for all operations, I would vote for removing it where it is useless.
> 
> I didn't test the procedure but I assume it is working ?
> 
> Thanks
> 
> Thomas

@thomasdanan @lucieleonard 
Across these three pages (Create an Account, Delete an Account, and Manage Access keys) we now have procedures to:

- Create Accounts (UI and CLI)
- Delete Accounts (UI)
- Create Access keys (UI and CLI)
- Delete Access keys(UI)

_Should we ask a Developer for:_

- Procedure to Delete Accounts via CLI
- Procedure to Delete Access Keys via CLI

**Comment by thomasdanan** [comment]
> > @lucieleonard @John-VanHook I have 2 global comments
> > 
> > 1. We should have a consistent organization of the information: Currently for each and every operation, you will get the UI way and the CLI way (within the same page). As such (if we are following this logic), generating Account Access Key in CLI should be under Managing Access Key Section
> > 2. I think it does not make sense to put the login page for all operations, I would vote for removing it where it is useless.
> > 
> > I didn't test the procedure but I assume it is working ?
> > Thanks
> > Thomas
> 
> @thomasdanan @lucieleonard
> Across these three pages (Create an Account, Delete an Account, and Manage Access keys) we now have procedures to:
> 
> * Create Accounts (UI and CLI)
> * Delete Accounts (UI)
> * Create Access keys (UI and CLI)
> * Delete Access keys(UI)
> 
> _Should we ask a Developer for:_
> 
> * Procedure to Delete Accounts via CLI
> * Procedure to Delete Access Keys via CLI

This is not really my point: Let's say I want to create an access key in CLI mode. I will go to Manage Access Key but I won't find the procedure because it is actually under Create Account operation.

What we are expecting is to have, for each operation, the UI and CLI way, side by side (on the same page). We may miss some CLI procedures for some operations but it is another problem.

**Comment by John-VanHook** [comment]
> > > @lucieleonard @John-VanHook I have 2 global comments
> > > 
> > > 1. We should have a consistent organization of the information: Currently for each and every operation, you will get the UI way and the CLI way (within the same page). As such (if we are following this logic), generating Account Access Key in CLI should be under Managing Access Key Section
> > > 2. I think it does not make sense to put the login page for all operations, I would vote for removing it where it is useless.
> > > 
> > > I didn't test the procedure but I assume it is working ?
> > > Thanks
> > > Thomas
> > 
> > 
> > @thomasdanan @lucieleonard
> > Across these three pages (Create an Account, Delete an Account, and Manage Access keys) we now have procedures to:
> > 
> > * Create Accounts (UI and CLI)
> > * Delete Accounts (UI)
> > * Create Access keys (UI and CLI)
> > * Delete Access keys(UI)
> > 
> > _Should we ask a Developer for:_
> > 
> > * Procedure to Delete Accounts via CLI
> > * Procedure to Delete Access Keys via CLI
> 
> This is not really my point: Let's say I want to create an access key in CLI mode. I will go to Manage Access Key but I won't find the procedure because it is actually under Create Account operation.
> 
> What we are expecting is to have, for each operation, the UI and CLI way, side by side (on the same page). We may miss some CLI procedures for some operations but it is another problem.

@thomasdanan @lucieleonard 
I changed the page structure(s) 
Is this what you mean ?

**Review by thomasdanan — APPROVED** [review]

**Comment by thomasdanan** [comment]
@John-VanHook perfect, thanks ... we are missing few CLI commands but that can come later.

File: docs/docsource/operation/account_operations/create_access_keys.rst [hunk 19]
```diff
@@ -0,0 +1,59 @@
...
+Create Access Keys Using the UI
+-------------------------------
+
+#. Log into the |product| UI using previously provisioned credentials (refer to :ref:`Create an Account`).
+
+#. Click **Accounts** on the top menu.
+
+   .. image:: ../../graphics/account_banner.PNG
+
+#. Click **+ Create Access key**
```
Comment (cathydossantospinto) [inline docs/docsource/operation/account_operations/create_access_keys.rst]:
```suggestion
#. Click **+ Create Access key**.
```

File: docs/docsource/operation/account_operations/create_access_keys.rst [hunk 20]
```diff
@@ -0,0 +1,59 @@
...
+
+   .. image:: ../../graphics/accounts-page.PNG
+
+#. Click **-> Continue** on the **Create Root user Access keys** screen.
+
+   .. image:: ../../graphics/root_user_access_keys_dialog.PNG
+
+#. Save the Secret Access key ID by clicking the copy button, and save it in a secure location
```
Comment (cathydossantospinto) [inline docs/docsource/operation/account_operations/create_access_keys.rst]:
```suggestion
#. Click the copy button to save the Secret Access key ID. Make sure to save it in a secure location.
```

File: docs/docsource/operation/account_operations/create_access_keys.rst [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+
+#. Click **-> Continue** on the **Create Root user Access keys** screen.
+
+   .. image:: ../../graphics/root_user_access_keys_dialog.PNG
+
+#. Save the Secret Access key ID by clicking the copy button, and save it in a secure location 
+   
+   .. image:: ../../graphics/close_button.PNG
+
+#. Close **Close** after you saved the Secret Access key ID.
```
Comment (cathydossantospinto) [inline docs/docsource/operation/account_operations/create_access_keys.rst]:
```suggestion
#. Click **Close** after saving the Secret Access key ID.
```

File: docs/docsource/operation/account_operations/create_access_keys.rst [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+
+Create Access Keys Using the Command Line
+-----------------------------------------
+
+#. Set the admin access key ID and secret access key ID as environment variables.
```
Comment (cathydossantospinto) [inline docs/docsource/operation/account_operations/create_access_keys.rst]:
```suggestion
#. Set the Admin Access key ID and Secret Access key ID as environment variables.
```

File: docs/docsource/operation/account_operations/create_access_keys.rst [hunk 23]
```diff
@@ -0,0 +1,59 @@
...
+
+   .. code::
```
Comment (cathydossantospinto) [inline docs/docsource/operation/account_operations/create_access_keys.rst]: I'd put these commands in [code-blocks](https://docs.google.com/document/d/1_N_-yrRGiygMxdJVhKdI8tEgCck50aNyH6UJsuEAnL4/edit#heading=h.kads838eo5z) for clarity. 
Caption 1 would be sth like `Vault Container`. Caption 2= `Admin Access key ID`. Caption 3= `Secret Access key ID`.

File: docs/docsource/operation/account_operations/create_access_keys.rst [hunk 24]
```diff
@@ -0,0 +1,59 @@
...
+
+      ADMIN_ACCESS_KEY_ID=$(kubectl --kubeconfig=/etc/kubernetes/admin.conf 
+      get secret artesca-data-management-vault-admin-creds.v1 -n zenko -o jsonpath='{.data.accessKey}' | base64 -d)
+
+   .. code::
+
+      ADMIN_SECRET_ACCESS_KEY=$(kubectl --kubeconfig=/etc/kubernetes/admin.conf 
+      get secret artesca-data-management-vault-admin-creds.v1 -n zenko -o jsonpath='{.data.secretKey}' | base64 -d)
+
+#. Generate the Account Access Key.
```
Comment (cathydossantospinto) [inline docs/docsource/operation/account_operations/create_access_keys.rst]:
```suggestion
#. Generate the Account Access key.
```

File: docs/docsource/operation/account_operations/create_an_account.rst [hunk 25]
```diff
@@ -34,31 +37,35 @@ not yet set up any accounts:
    click the **Locations** tab to manage account locations and the buckets they
    contain.
    
-Create an Account from the Command Line
----------------------------------------
+Create an Account Using the Command Line
+----------------------------------------
 
-To create an account from the command line:
+#. Set the admin access key ID and secret access key ID as environment variables.
```
Comment (cathydossantospinto) [inline docs/docsource/operation/account_operations/create_an_account.rst]:
```suggestion
#. Set the Admin Access key ID and Secret Access key ID as environment variables.
```

File: docs/docsource/operation/account_operations/create_access_keys.rst [hunk 26]
```diff
@@ -0,0 +1,59 @@
...
+
+   .. note:: 
+
+      In the following example, account name is ``account1`` and email is ``account1@scality.local``
```
Comment (cathydossantospinto) [inline docs/docsource/operation/account_operations/create_access_keys.rst]:
```suggestion
      In the following example, account name is ``account1`` and email is ``account1@scality.local`` .
```

File: docs/docsource/operation/account_operations/create_an_account.rst [hunk 27]
```diff
@@ -34,31 +37,35 @@ not yet set up any accounts:
...
 
-#. Retrieve tokens as described in :ref:`Retrieve Access Tokens`.
+   .. code::
```
Comment (cathydossantospinto) [inline docs/docsource/operation/account_operations/create_an_account.rst]: Ditto for the code-blocks 😉

File: docs/docsource/operation/account_operations/delete_access_keys.rst [hunk 28]
```diff
@@ -0,0 +1,21 @@
...
+Delete Access Keys Using the UI
+-------------------------------
+
+#. Log into the |product| UI.
+
+#. Click **Accounts** on the top menu.
+
+   .. image:: ../../graphics/account_banner.PNG
+
+#. Click the red trash icon to the right of the access key you wish to delete.
```
Comment (cathydossantospinto) [inline docs/docsource/operation/account_operations/delete_access_keys.rst]:
```suggestion
#. Click the red trash icon next to the Access key you want to delete. 
```

File: docs/docsource/operation/account_operations/delete_an_account.rst [hunk 29]
```diff
@@ -3,6 +3,9 @@
 Delete an Account
 =================
 
+Delete an Account Using the UI
+------------------------------
+
 To delete an account:
```
Comment (cathydossantospinto) [inline docs/docsource/operation/account_operations/delete_an_account.rst:9]: Useless sentence. Can be removed.

**Review by cathydossantospinto — APPROVED** [review]

**Comment by cathydossantospinto** [comment]
Wake up bert-e!!

**Pull request merged**